crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -1952,6 +1974,49 @@ type SharedInput {
 	mutable: Boolean!
 }
 
+"""
+The transaction accepted a shared object as input, but it was deleted before the transaction
+executed.
+"""
+type SharedObjectDelete {
+	"""
+	Address of the shared object.
+	"""
+	address: SuiAddress!
+	"""
+	The version of the shared object that was assigned to this transaction during by consensus,
+	during sequencing.
+	"""
+	version: Int!
+	"""
+	Whether this transaction intended to use this shared object mutably or not.
```

Comment:
I do not really understand what `intended` actually entails. If this field is true, what does that mean to this shared object?

Reply:
In this situation we are talking about a transaction that failed because an object it requested during signing was deleted before it executed.  This flag tells us whether the transaction intended to just read the shared object, or modify it.  This has implications for consensus: Transactions that just read shared objects at a certain version can be executed concurrently, while transactions that write shared objects must be executed serially with respect to each other.

Reply:
Appreciate taking the time to answer. Particularly, the last sentence is the one that made everything click!

Reply:
Okay great!  I'll add some documentation to the `SharedInput` type with the content of that last sentence, and refer to it from here.